packages/a-msgpack/src/Encoder.ts

```diff
@@ -359,7 +359,14 @@ export class Encoder {
   }
 
   encodeExtension(ext: ExtData): void {
-    const size = ext.data.length;
+    this.encodeExtensionHeader(ext);
+    if (ext.data !== WRITE_ONLY_HEADER) {
+      this.encodeExtensionData(ext.data);
+    }
+  }
+
+  encodeExtensionHeader(ext: ExtData): void {
+    const size = ext.data === WRITE_ONLY_HEADER ? 0 : ext.data.length;
```

Comment:
why not always use ext.data.length if you were to force somehow not allowing ext.data to be null?